scapy/layers/ipsec.py

```diff
@@ -165,6 +165,270 @@ def data_for_encryption(self):
         return raw(self.data) + self.padding + struct.pack("BB", self.padlen, self.nh)  # noqa: E501
 
 
+###############################################################################
+if conf.crypto_valid_advanced:
+    from cryptography.exceptions import InvalidTag
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.ciphers import (
+        aead,
+    )
+else:
+    log_loading.info("Can't import python-cryptography v1.7+. "
+                     "Disabled IPsec encryption/authentication.")
+    default_backend = None
+    InvalidTag = Exception
+    aead = None
```

Comment:
Could you merge those imports with the ones at the top?

Reply:
Hello,
Moved those imports to the top done. Do we need to do that also for crypt and auth algo (same logic here)?

Reply:
Same coding style as crypt and auth classes. Do I need to merge all imports at the top of the file?

Reply:
Pending actions discussion before doing changes.